src/TraceEvent/TraceLog.cs

```diff
@@ -51,27 +51,49 @@ namespace Microsoft.Diagnostics.Tracing.Etlx
     /// </summary>
     public sealed class TraceLog : TraceEventSource, IDisposable, IFastSerializable, IFastSerializableVersion
     {
+        /// <summary>
+        /// Given an array of files, create an ETLX file for the data. 
+        /// <para>If etlxFilePath is null the output name is derived from etlFilePath by changing its file extension to .ETLX.</para>
+        /// <returns>The name of the ETLX file that was generated.</returns>
+        /// </summary>
+        public static string CreateFromEventTraceLogFiles(IEnumerable<string> fileNames, string etlxFilePath = null, TraceLogOptions options = null, TraceEventDispatcherOptions traceEventDispatcherOptions = null)
```

Comment:
I don't think you should create this method, but instead use the other one that you created that takes a dispatcher directly.